src/sage/functions/gamma.py

```diff
@@ -197,75 +205,76 @@ def __init__(self):
         Numerical evaluation happens when appropriate, to the
         appropriate accuracy (see :trac:`10072`)::
 
-            sage: log_gamma(6)
+            sage: log_gamma(6)                                                          # needs sage.symbolic
             log(120)
-            sage: log_gamma(6.)
+            sage: log_gamma(6.)                                                         # needs sage.symbolic
             4.78749174278205
```

Comment:
no `mpmath`?

Reply:
`log_gamma` is a `GinacFunction`, so it unconditionally goes through the symbolic subsystem. Not sure how it is evaluated in the end; I don't have a distribution that would test symbolics without `mpmath`

Reply:
You mean the distribution that installs the feature `sage.symbolics` also installs `mpmath`. I confirmed this with `sagemath-symbolics`.

So (1) `bessel_Y` needs only `mpmath` for a numerical input while (2) `log_gamma` needs full `sage.symbolic` for a numerical input. This difference comes from the implementation details.

Adding `needs` tag based on this much implementation details seems very bad. From the point of view of a user using `bessel_Y`, she needs to be informed of what features should be available in her installation to use `bessel_Y`. She would want to think of this in terms of only `bessel_Y`, not of "`bessel_Y` with a numerical input` or "`bessel_Y` with a symbolic input`. Anyway suppose she was informed about this (1). Then if she also wants to use `log_gamma`, she would be surprised that what she learned about `bessel_Y` is not transferred to `log_gamma` (2).

I think `bessel_Y(0.5, 1.5)` should also be tagged with `# needs sage.symbolic` even though `sage.symbolic` is not the minimum requirement. So the user would just think remember that to use `bessel_Y`, she needs `sage.symbolic`.

Reply:
> [...] I think `bessel_Y(0.5, 1.5)` should also be tagged with `# needs sage.symbolic` even though `sage.symbolic` is not the minimum requirement.

I have a different view on this. A major point of this PR is to make it possible to numerically evaluate functions without the symbolic subsystem present. This goal is already achieved for `BuiltinFunction`s (such as `bessel_Y`). The doctests for these functions document this.

That the goal is not achieved yet for `GinacFunction`s (such as `log_gamma`) should be viewed just as a current implementation restriction, which we should try to remove in a later PR. The doctests that have `# needs sage.symbolic` for numerical input document the status quo.

Reply:
OK. Then she would just remember that she needs `mpmath` to use `bessel_Y` and `log_gamma` (this only later) unless the input is symbolic.